hyprpm/CMakeLists.txt

```diff
@@ -13,6 +13,17 @@ pkg_check_modules(hyprpm_deps REQUIRED IMPORTED_TARGET tomlplusplus hyprutils>=0
 
 add_executable(hyprpm ${SRCFILES})
 
+# glaze
+include(FetchContent)
+FetchContent_Declare(
+  glaze
+  GIT_REPOSITORY https://github.com/stephenberry/glaze.git
+  GIT_TAG v4.2.2
+  GIT_SHALLOW TRUE
+)
+FetchContent_MakeAvailable(glaze)
+target_link_libraries(hyprpm PRIVATE glaze::glaze)
```

Comment:
glaze is in the AUR: https://repology.org/project/glaze/versions

Do not do this, just add a dep in pkg-config.

Hyprland is a big enough project to not worry about glaze not being widely packaged, because we can force packagers to package it. :)

Reply:
How would you fix the CI/CD, then? Should I add a step to the setup_base action to compile glaze from source?

In the latest commit, I tried adding it as a submodule which works for cmake but not yet for meson, haven't figured out how to do that yet. But I guess I'll just remove it and do it with pkg-config instead.

Reply:
Had to use find_package cause glaze doesn't supply a .pc file.

Reply:
> How would you fix the CI/CD, then? Should I add a step to the setup_base action to compile glaze from source?

CI/CD already pulls from git some dependencies. Just add a new one.

> In the latest commit, I tried adding it as a submodule which works for cmake but not yet for meson, haven't figured out how to do that yet. But I guess I'll just remove it and do it with pkg-config instead.

you have to ping @fufexan 3 times at 2am and he will appear behind you and fix meson

Reply:
meson seems to work now with the system-installed glaze.
But I'm having issues with the Nix build and I have 0 experience with that. Who do I conjure up for that? :smile:

Reply:
fufexan as well